lib/features/popup-menu/PopupMenuHeader.js

```diff
@@ -0,0 +1,91 @@
+import classNames from 'clsx';
+
+import {
+  html,
+  useMemo
+} from '../../ui';
+
+/**
+ * @typedef {import('./PopupMenuProvider').PopupMenuHeaderEntry} PopupMenuHeaderEntry
+ */
+
+/**
+ * Component that renders a popup menu header.
+ *
+ * @param {Object} props
+ * @param {PopupMenuHeaderEntry[]} props.headerEntries
+ * @param {PopupMenuHeaderEntry} props.selectedEntry
+ * @param {(event: MouseEvent, entry: PopupMenuHeaderEntry) => void} props.onSelect
+ * @param {(entry: PopupMenuHeaderEntry | null) => void} props.setSelectedEntry
+ * @param {string} props.title
+ */
+export default function PopupMenuHeader(props) {
+  const {
+    headerEntries,
+    onSelect,
+    selectedEntry,
+    setSelectedEntry,
+    title
+  } = props;
+
+  const groups = useMemo(() => groupEntries(headerEntries), [ headerEntries ]);
+
+  return html`
+    <div class="djs-popup-header">
+      <h3 class="djs-popup-title" title=${ title }>${ title }</h3>
+      ${ groups.map((group, index) => html`
+        ${ index > 0 && html`<div class="divider" />` }
```

Comment:
Works on my machine; only thing I see is that adding the divider is different across vertical entries (organized in `div` groups) and horizontal header entries (organized in flat list).

Did you consider stacking in `div` for horizontal grouping?

Reply:
No I didn't. Let me me look into that.